summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -0,0 +1,103 @@
+package com.twitter.summingbird.storm.collector
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.online.option.SummerBuilder
+import backtype.storm.spout.SpoutOutputCollector
+import com.twitter.algebird.util.summer.AsyncSummer
+import com.twitter.util.{ Await, Future }
+import java.util.{ List => JList }
+import scala.collection.mutable.{ Map => MMap }
+import scala.collection.mutable.{ MutableList => MList }
+import scala.collection.{ Map => CMap }
+import scala.collection.JavaConverters._
+import java.util.{ List => JList }
+
+class AggregatorOutputCollector[K, V: Semigroup](in: SpoutOutputCollector, func: JList[AnyRef] => JList[AnyRef], summerBuilder: SummerBuilder, summerShards: KeyValueShards) extends TransformingOutputCollector(in, func) {
+
+  var spoutCaches = MMap[String, AsyncSummer[(K, V), Map[K, V]]]()
+  var lastDump = System.currentTimeMillis()
+  var streamMessageIdTracker = MMap[String, MMap[Int, MList[Object]]]()
+
+  def timerFlush() = {
+    spoutCaches.keys.foreach { stream =>
+      val tupsOut = spoutCaches(stream).tick.map { convertToSummerInputFormat(_) }
+      val tups = Await.result(tupsOut)
+
+      tups.foreach {
+        case (k, v) => {
+          val messageIdsTracker = streamMessageIdTracker(stream)
+          val messageIds = messageIdsTracker.remove(k)
+          (messageIds.isEmpty, stream.isEmpty) match {
+            case (true, true) => in.emit(List(k, v).asJava.asInstanceOf[JList[AnyRef]])
```

Comment:
We're doing this List construction in all cases.  Let's just do it once ahead of time.  Might make sense to construct it directly in Java too.